openedx/core/djangoapps/user_authn/views/tests/test_reset_password.py

```diff
@@ -713,3 +713,87 @@ def test_reset_password_token_with_other_user(self):
 
         self.user = User.objects.get(pk=self.user.pk)
         self.assertFalse(self.user.is_active)
+
+
+@ddt.ddt
+@unittest.skipUnless(
+    settings.ROOT_URLCONF == "lms.urls",
+    "reset password tests should only run in LMS"
+)
+class ResetPasswordAPITest(CacheIsolationTestCase):
+    """Tests of the logistration API's password reset endpoint. """
+    request_factory = RequestFactory()
+    ENABLED_CACHES = ['default']
+
+    def setUp(self):
+        super(ResetPasswordAPITest, self).setUp()
+        self.user = UserFactory.create()
+        self.user.save()
+        self.token = default_token_generator.make_token(self.user)
+        self.uidb36 = int_to_base36(self.user.id)
+
+    @ddt.data(
+        (None, None, True),
+        (None, 'invalid_token', False),
+    )
+    @ddt.unpack
+    def test_invalid_token_in_password_reset_request(self, uidb36, token, status):
+        """
+         Test that user should not be able to reset password through invalid token
+        """
+        if uidb36 is None:
+            uidb36 = self.uidb36
+        if token is None:
+            token = self.token
+
+        request_param = {'new_password1': 'new_password1', 'new_password2': 'new_password1'}
+        post_request = self.request_factory.post(
+            reverse(
+                "logistration_password_reset",
+                kwargs={"uidb36": uidb36, "token": token}
+            ),
+            request_param
+        )
+        process_request(post_request)
+        post_request.user = AnonymousUser()
+        json_response = password_reset_logistration(post_request, uidb36=uidb36, token=token)
+        json_response = json.loads(json_response.content.decode('utf-8'))
```

Comment:
Can we move this into a helper function and use here and in `test_password_mismatch_in_reset_request`? Seems same code to me except the second password value. We can use a second password keyword argument with a default value `new_password1` and pass a different value from other test case.